newrelic-weaver/src/test/java/com/newrelic/weave/weavepackage/WeavePackageManagerTest.java

```diff
@@ -239,7 +239,8 @@ public void testMaxInvalidClassLoaders() throws IOException {
             wpm.weave(cl, "com/newrelic/weave/weavepackage/WeavePackageManagerTest$NewNoMatchClass",
                     WeaveTestUtils.getClassBytes("com.newrelic.weave.weavepackage.WeavePackageManagerTest$NewNoMatchClass"));
         }
-        Assert.assertTrue(WeavePackageManager.MAX_INVALID_PACKAGE_CACHE >= wpm.invalidPackages.size());
+        wpm.invalidPackages.cleanUp();
```

Comment:
See the [wiki](https://github.com/ben-manes/caffeine/wiki/Guava); Caffeine's size-based eviction may cause the cache to grow _slightly_ larger than configured. Here we trigger an explicit cleanup to make sure the test passes. Under normal operation this kind of cleanup is performed in the context of subsequent cache operations.

Reply:
This one `.cleanUp()` call wasn't enough; added a commit with a more robust fix.